dask/array/tests/test_overlap.py

```diff
@@ -840,3 +842,15 @@ def test_map_overlap_new_axis():
     # Shape and chunks aren't known until array is computed,
     # so don't expclitly check shape or chunks in assert_eq
     assert_eq(expected, actual, check_shape=False, check_chunks=False)
+
+
+def test_overlap_now_blowing_up_graph():
```

Comment:
```suggestion
def test_overlap_not_blowing_up_graph():
```

Reply:
Also, is there a better way to test this with overlap directly? This test will become out-of-sync should the Xarray implementation change.

Reply:
We could, yes. But I'd rather make sure that xarray isn't changing the way they call map_overlap to avoid them running into the error again